bsb/storage/interfaces.py

```diff
@@ -930,7 +930,9 @@ def all(self):
         for len_, local_, global_ in zip(lens, locals_, globals_):
             lloc[ptr : ptr + len_] = local_
             gloc[ptr : ptr + len_] = global_
-        return lcol, lloc, gcol, gloc
+            ptr += len_
+        result = lcol, lloc, gcol, gloc
+        yield result
```

Comment:
```suggestion
        return lcol, lloc, gcol, gloc
```

No need for `all()` to be a generator, a standard return value should do and work as expected in more circumstances.

Reply:
Using return still gives the error 
`ValueError: too many values to unpack (expected 4)`
calling CSIterator.all() and trying to unpack its data, for example with the following code:

```
cs = self.scaffold.get_connectivity_set("mossy_fibers_to_glomerulus") 
iter = cs.from_(pre_ps.get_loaded_chunks()).to(post_ps.get_loaded_chunks())
for local_chunk_ids, local_locs, global_chunk_ids, global_locs in iter.all():
...
```
Instead, if the method all() is a generator the code above works fine. :shrug:

Reply:
The suggestion is correct, the use of `all()` i n a for loop is incorrect: there is only 1 result from `all`:

```python
local_chunk_ids, local_locs, global_chunk_ids, global_locs in iter.all()
```

Reply:
Right, there is only one result. There is no need for a loop.